cirq-google/cirq_google/calibration/phased_fsim.py

```diff
@@ -1038,3 +1046,51 @@ def try_convert_sqrt_iswap_to_fsim(gate: cirq.Gate) -> Optional[PhaseCalibratedF
         return PhaseCalibratedFSimGate(cirq.FSimGate(theta=np.pi / 4, phi=0.0), 0.5)
 
     return None
+
+
+def try_convert_gate_to_fsim(gate: cirq.Gate) -> Optional[PhaseCalibratedFSimGate]:
+    """Converts a gate to equivalent PhaseCalibratedFSimGate if possible.
+
+    Args:
+        gate: Gate to convert.
+
+    Returns:
+        If provided gate is equivalent to some PhaseCalibratedFSimGate, returns that gate.
+        Otherwise returns None.
+    """
+    phi = 0.0
+    theta = 0.0
+    phase_exponent = 0.0
+    if isinstance(gate, SycamoreGate):
+        phi = np.pi / 6
+        theta = np.pi / 2
+    elif isinstance(gate, cirq.FSimGate):
+        theta = gate.theta
+        phi = gate.phi
+    elif isinstance(gate, cirq.ISwapPowGate):
+        if not np.isclose(np.exp(np.pi * 1j * gate.global_shift * gate.exponent), 1.0):
+            return None
+        theta = -gate.exponent * np.pi / 2
+    elif isinstance(gate, cirq.PhasedFSimGate):
+        if not np.isclose(gate.zeta, 0.0) or not np.isclose(gate.gamma, 0.0):
+            return None
+        theta = gate.theta
+        phi = gate.phi
+        phase_exponent = -gate.chi / (2 * np.pi)
+    elif isinstance(gate, cirq.PhasedISwapPowGate):
+        theta = -gate.exponent * np.pi / 2
+        phase_exponent = -gate.phase_exponent
+    elif isinstance(gate, cirq.ops.CZPowGate):
+        if not np.isclose(np.exp(np.pi * 1j * gate.global_shift * gate.exponent), 1.0):
+            return None
+        phi = -np.pi * gate.exponent
+    else:
+        return None
+
+    phi = phi % (2 * np.pi)
+    theta = theta % (2 * np.pi)
+    if theta >= np.pi:
+        theta = 2 * np.pi - theta
+        phase_exponent = phase_exponent + 0.5
+    phase_exponent %= 1
```

Comment:
This assumes that the gate to convert is not parameterized, is this assumption correct? If yes, please document it and raise error if not valid. If not, then please update the modular arithmetic to use https://github.com/quantumlib/Cirq/blob/0311e1746acdb4519f334c421a90e15ff3035891/cirq-google/cirq_google/common_serializers.py#L51

Reply:
Hi. I didn't realize that some gates have cirq.value.TParamVal as theoir parameters. I assume you are referring to that.

Since output of this function (PhaseCalibratedFSimGate and FSimGate) can have only `float` parameters, we can't support parameterized gates.

I'm going to add a check that all parameters are `float`s, and otherwise return `None`.

Reply:
The implementation of `FSim` gate does allow the parameters to be symbols (though the types in constructor are not correctly updated).  Eg:

https://github.com/quantumlib/Cirq/blob/12530f1531e5da663a6b172b1967763e08819912/cirq-core/cirq/ops/fsim_gate.py#L39

Hence, the following code is perfectly valid in Cirq. 

```python
fsim = cirq.FSimGate(theta = sympy.Symbol('theta'), phi = sympy.Symbol('phi'))
cirq.Circuit(fsim(*cirq.LineQubit.range(2)))
```

I guess what you mean is that for the specific context here, you do not wish to support parameterized gates? You can use `cirq.is_parameterized()` protocol to check whether the input gate is parameterized or not.

Reply:
Thanks, I used `cirq.is_paramterized()`.